scheduler/src/main/java/org/apache/falcon/workflow/engine/FalconWorkflowEngine.java

```diff
@@ -108,12 +108,17 @@ public void dryRun(Entity entity, String clusterName, Boolean skipDryRun) throws
 
     @Override
     public boolean isActive(Entity entity) throws FalconException {
-        EntityID id = new EntityID(entity);
-        // Ideally state store should have all entities, but, check anyway.
-        if (STATE_STORE.entityExists(id)) {
-            return STATE_STORE.getEntity(id).getCurrentState() != EntityState.STATE.SUBMITTED;
+        try {
+            EntityID id = new EntityID(entity);
+            // Ideally state store should have all entities, but, check anyway.
+            if (STATE_STORE.entityExists(id)) {
+                return STATE_STORE.getEntity(id).getCurrentState() != EntityState.STATE.SUBMITTED;
+            }
+            return false;
+        } catch (NullPointerException npe) {
```

Comment:
May be we should do a refactoring of this for better handling of the case when JPAService is not there in a future patch.

+1 for the current one